workspace/lib/mix/tasks/workspace.list.ex

```diff
@@ -160,13 +167,19 @@ defmodule Mix.Tasks.Workspace.List do
   end
 
   defp list_or_save_workspace_projects(workspace, opts) do
-    case opts[:json] do
-      false -> list_workspace_projects(workspace, opts[:show_status])
-      true -> write_json(workspace, opts)
-    end
+    # remove once we drop support for `--json`
```

Comment:
```suggestion
    # TODO: remove --json support in 0.4.0
```

(to help me grep this when it's time)